pkg/scheduling/scheduling_test.go

```diff
@@ -0,0 +1,137 @@
+package scheduling
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+)
+
+// Monkey patching is used here
+func TestProcessPidsCPUScheduling(t *testing.T) {
+	testPids := []int{101, 102}
+	testContainer := &provider.Container{}
+
+	testCases := []struct {
+		mockGetProcessCPUScheduling           func(int, *provider.Container) (string, int, error)
+		check                                 string
+		expectedCPUSchedulingConditionSuccess bool
+	}{
+		{
+			mockGetProcessCPUScheduling: func(pid int, container *provider.Container) (string, int, error) {
+				return "SCHED_OTHER", 0, nil
+			},
+			check:                                 SharedCPUScheduling,
+			expectedCPUSchedulingConditionSuccess: true,
+		},
+		{
+			mockGetProcessCPUScheduling: func(pid int, container *provider.Container) (string, int, error) {
+				return "SCHED_RR", 90, nil
+			},
+			check:                                 SharedCPUScheduling,
+			expectedCPUSchedulingConditionSuccess: false,
+		},
+		{
+			mockGetProcessCPUScheduling: func(pid int, container *provider.Container) (string, int, error) {
+				return "SCHED_FIFO", 9, nil
+			},
+			check:                                 ExclusiveCPUScheduling,
+			expectedCPUSchedulingConditionSuccess: true,
+		},
+		{
+			mockGetProcessCPUScheduling: func(pid int, container *provider.Container) (string, int, error) {
+				return "SCHED_FIFO", 11, nil
+			},
+			check:                                 ExclusiveCPUScheduling,
+			expectedCPUSchedulingConditionSuccess: false,
+		},
+	}
+	for _, tc := range testCases {
+		GetProcessCPUSchedulingFn = tc.mockGetProcessCPUScheduling
```

Comment:
You should save the original func in a variable then after the test are complete, set it back to the original.  For example: https://github.com/test-network-function/cnf-certification-test/blob/e4a631bb3e91945af53f31e7195622a9ee3eb15b/cnf-certification-test/lifecycle/scaling/deployment_scaling_test.go#L61